Github/service.js

```diff
@@ -1,134 +1,164 @@
+    request.get(
+      {
+        url: repoUrl,
+        headers: { 'User-Agent': 'request' },
+      },
+      (err, res, data) => {
+        if (err) {
+          console.log('Error:', err);
+          reject(new Error(`Unable to get repos.`));
+        } else {
+          const reposX = JSON.parse(data);
+          reposX.forEach(repo => {
+            repo.branches_url = repo.branches_url.slice(
+              0,
+              repo.branches_url.length - 9,
+            );
+            repo.branches_url = `${
+              repo.branches_url
+              }?per_page=100&access_token=${key}`;
+            if (
+              new Date(today - new Date(repo.pushed_at)) < new Date(recency)
+            ) {
+              branchUrls.push({
+                name: repo.name,
+                url: repo.branches_url,
+              });
             }
-        });
-    });
-}
+          });
+          resolve();
+        }
+      },
+    );
+  });
+};
 
-const getCommits = (commitObject) => {
-    return new Promise((resolve, reject) => {
-        request.get({
-            url: 'https://api.github.com/repos/Seneca-CDOT/' + commitObject.repo + '/commits?sha=' + commitObject.br.sha + '&per_page=100&access_token=' + key,
-            headers: { 'User-Agent': 'request' },
-        }, (err, res, data) => {
-            if (err) {
-                console.log('Error:', err);
-                reject(err);
-            } else {
-                JSON.parse(data).forEach(singleCommit => {
-                    if (new Date(today - new Date(singleCommit.commit.author.date)) < new Date(recency)) {
-                        // To make it work with Dashboard/src/components/Github.js
-                        const time = new Date(singleCommit.commit.author.date);
-                        singleCommit.author.date = time.toLocaleString("en-US", {timeZone : "America/Toronto"});
-                        singleCommit.author.name = singleCommit.commit.author.name;
-                        singleCommit.message = singleCommit.commit.message;
-                        singleCommit.repoName = commitObject.repo;
-                        singleCommit.branchName = commitObject.br.name;
-                        recentCommits.push(singleCommit);
-                    }
-                });
-                resolve();
+const getCommits = commitObject => {
+  return new Promise((resolve, reject) => {
+    request.get(
+      {
+        url: `https://api.github.com/repos/Seneca-CDOT/${
+          commitObject.repo
+        }/commits?sha=${commitObject.br.sha}&per_page=100&access_token=${key}`,
+        headers: { 'User-Agent': 'request' },
+      },
+      (err, res, data) => {
+        if (err) {
+          console.log('Error:', err);
+          reject(err);
+        } else {
+          JSON.parse(data).forEach(singleCommit => {
+            if (
+              new Date(today - new Date(singleCommit.commit.author.date)) <
+              new Date(recency)
```

Comment:
We can improve this. Subtracting two Date objects results in a number which means we don't have to wrap it in another Date object nor do we need to convert recency to a Date. i.e.

`if ( today - new Date(author.date) < recency)`